### PR TITLE
added option to remove old folder after moving to new location

### PR DESCRIPTION
**Pull request by frwny**
as above - instead of copying the folder it behaves like a rename

i would argue that this would be a sensible default for the script due to the semantics of "rename" in the prompt. however i've added it as a flag for now to avoid breaking changes 

please double check this as its my first pr into this repo x

### CODE REVIEW

File: salmon/uploader/__init__.py [hunk 1]
```diff
@@ -154,6 +154,11 @@
     is_flag=True,
     help='Skip check for MQA marker (on first file only)',
 )
+@click.option(
+    "--move-folder",
```
Comment (digerati-red) [inline salmon/uploader/__init__.py:158]:
```suggestion
    "--remove-folder",
```

Comment (digerati-red) [reply salmon/uploader/__init__.py:158]: If the functionality is to remove the folder, the verbiage should be consistent IMO

**Review by digerati-red — CHANGES_REQUESTED** [review]

**Comment by digerati-red** [comment]
> i would argue that this would be a sensible default for the script due to the semantics of "rename" in the prompt.

It's worth discussing. My understanding currently is intention of this rename and related functionality is to try and leave the source folder generally as is.  The output to the `DOWNLOAD_DIRECTORY` only has any renaming modifications done from salmon

https://github.com/smokin-salmon/smoked-salmon/blob/fdaeccc53baea255e8ac9b2debb6d22af7ece136/salmon/tagger/foldername.py#L49
`new_path` is comprised of the `config.DOWNLOAD_DIRECTORY`, so renaming is happening to the output folder. Currently the source folder is not renamed or removed.

Seems like the flag would either be something like: 

`--rename-source-folder` - If salmon renames folder, also rename source
`--remove-source-folder` - After copying to the download directory, remove the source folder

**Comment by redusys** [comment]
> `--remove-source-dir` - After copying to the download directory, remove the source folder

I am also in favor of the `--remove-source-dir` naming proposal. It feels consistent with the rest of the tool.

Just to share a bit of my understanding of smoked-salmon's philosophy (based on reading the source over and over in the past few weeks): I believe the original intent was that albums are first downloaded to a location outside the torrent client's seeding directories (let's call it location A).

Then, when using smoked-salmon:
- The album gets tagged in place in location A, so the source directory is actually modified during this step (except if using the `--scene` flag).
- After tagging, the album is copied to config.DOWNLOAD_DIRECTORY, which should be the seeding directory for the torrent client. This enables automatic torrent addition (e.g., via a watch folder in qBittorrent) so that seeding begins right away. This is during this copy that the folder/files are actually renamed (they are copied to their new names)

With this in mind, I think `--remove-source-dir` is a fitting and coherent name for the flag.

There are some edge cases to test, like avoid removing if the source dir is the same as the destination dir (for people, like me, who put everything into the config.DOWNLOAD_DIRECTORY, even before upload). This could happen in case of a `--scene`, or if the folder is by chance already using the right naming. I will have a look at the code of this PR tomorrow.

**Comment by digerati-red** [comment]
> I believe the original intent..

+1

> like avoid removing if the source dir is the same as the destination dir (for people, like me, who put everything into the config.DOWNLOAD_DIRECTORY, even before upload

Does that fit this case?

https://github.com/smokin-salmon/smoked-salmon/blob/095b6135e5c6d12f9178f6a1def95722ff366c28/salmon/tagger/foldername.py#L65

File: salmon/tagger/foldername.py [hunk 2]
```diff
@@ -67,6 +67,8 @@ def rename_folder(path, metadata, auto_rename, check=True):
     else:
         shutil.copytree(path, new_path, dirs_exist_ok=True)
         click.secho(f"Copied folder to '{new_path}'.", fg="yellow")
+        if rename_folder:
```
Comment (redusys) [inline salmon/tagger/foldername.py:70]: I think you want to test `move_folder` (which is passed as parameter to this function).
Or if we go along with the naming proposed by @digerati-red , the variable should be renamed to `remove-source-dir`

**Review by redusys — DISMISSED** [review]

**Comment by redusys** [comment]
> > like avoid removing if the source dir is the same as the destination dir (for people, like me, who put everything into the config.DOWNLOAD_DIRECTORY, even before upload
> 
> Does that fit this case?

Yes, I actually remember adding that for the `--scene` releases I was testing against 😄 
So only the renaming of the option + variable check, and a nice rebase, and we could merge this PR.

I am also in favor of leaving this option as False by default, but open to other opinions. Thoughts?

**Comment by digerati-red** [comment]
> I am also in favor of leaving this option as False by default
+1

I personally use a working folder, where I copy from my permanent collection to, then remove the working folders. This enhancement is def an upgrade for my current process.

That said, once I'm 💯 comfortable salmon isn't going to modify anything I don't want it to... I could see running this on my permanent collection and just making necessary folder name changes to the copied output folders.

If nothing else, current functionality doesn't remove so I say it should be default False for now

**Comment by frwny** [comment]
Moved the parameter from a runtime flag to a config option and rebased onto the new HEAD as requested

File: salmon/tagger/foldername.py [hunk 3]
```diff
@@ -67,6 +67,9 @@ def rename_folder(path, metadata, auto_rename, check=True):
     else:
         shutil.copytree(path, new_path, dirs_exist_ok=True)
         click.secho(f"Copied folder to '{new_path}'.", fg="yellow")
+        if config.REMOVE_SOURCE_DIR:
+            if path != new_path:
```
Comment (redusys) [inline salmon/tagger/foldername.py:71]: We can remove this check actually, @digerati-red pointed out that this case is already taken care of line 65 of the same file (and in a safer way actually, using os.path.samefile())

**Review by redusys — CHANGES_REQUESTED** [review]
Left a comment with a suggestion to simplify the PR a bit more 🙂

Also, one quick favor: could you add this option to `config.py.txt` as well? I think it fits under `UPLOADING & PROCESSING SETTINGS`. I added all the options there a few commits back so it's easier for users to see everything available in smoked salmon.

**Comment by redusys** [comment]
Can't wait to get this PR merged, removing the old folders manually has been slightly annoying until now 💯

**Comment by redusys** [comment]
> Added documentation now. Do you want me to rebase and squash these all into 1 commit before merging?

That would be awesome! The magic of rebase/squash is that it lets us pretend we nailed everything perfectly from the start 😄 If only real life worked like that!

**Comment by frwny** [comment]
> > Added documentation now. Do you want me to rebase and squash these all into 1 commit before merging?
> 
> That would be awesome! The magic of rebase/squash is that it lets us pretend we nailed everything perfectly from the start 😄 If only real life worked like that!

Sweet all squashed and ready :)

**Review by redusys — APPROVED** [review]
